api/data/seattle_db.go

```diff
@@ -11,16 +11,31 @@ import (
 
 // SeattleOfficer is the object model for SPD officers
 type SeattleOfficer struct {
-	Date            time.Time    `json:"date,omitempty"`
-	Badge           string       `json:"badge,omitempty"`
-	FullName        string       `json:"full_name,omitempty"`
-	Title           string       `json:"title,omitempty"`
-	Unit            string       `json:"unit,omitempty"`
-	UnitDescription nulls.String `json:"unit_description,omitempty"`
-	FirstName       string       `json:"first_name,omitempty"`
-	MiddleName      nulls.String `json:"middle_name,omitempty"`
-	LastName        string       `json:"last_name,omitempty"`
-	Current         bool         `json:"is_current"`
+	Date            string `json:"date,omitempty"`
+	Badge           string `json:"badge,omitempty"`
+	FullName        string `json:"full_name,omitempty"`
+	Title           string `json:"title,omitempty"`
+	Unit            string `json:"unit,omitempty"`
+	UnitDescription string `json:"unit_description,omitempty"`
+	FirstName       string `json:"first_name,omitempty"`
+	MiddleName      string `json:"middle_name,omitempty"`
+	LastName        string `json:"last_name,omitempty"`
+	Current         bool   `json:"is_current"`
```

Comment:
What is the difference between `nulls.String` and `string`? And what is this `` `json...` `` magic that is happening?

Reply:
`nulls.String` handles any fields that may be null in the db, while string expects a non null field

the `json:` tag tells go which field in json gets assigned to which field in the struct. In this case, the `Current` struct field corresponds to the `is_current` field in json